### ISSUE TITLE
Oak sometimes crashes due to duplicate lifecycle instances starting

### ISSUE THREAD

**Issue by 200sc** [oakmound/oak#3]
There are some very rare crashes we've seen in Oak that we haven't been able to duplicate, but there is one crash in particular that happens fairly frequently (1/20 runs) which terminates the engine as soon as it starts up, reported by the error [here](https://github.com/OakmoundStudio/oak/blob/master/lifecycle.go#L30). Sometimes this error happens and the engine runs normally, other times it happens and the engine stops. 

We believe this is a shiny bug, but we can't track down where it's coming from in shiny if it is. It may be Windows specific. It seems to happen more frequently on Win32 than Win64.

**Comment by 200sc** [comment]
If we were to move all of the data fields that are global in the oak package to a controller struct (and also remove legacy support, and replace some things in the event package) then we would be able to deal with this issue differently by actually allowing multiple instances of the engine to be started. 

That wouldn't solve this issue, as this was something we discovered because win32 would spawn three windows when calling Init, but it would maybe mitigate it. 

Dealing with console input would be another issue with that change.

**Comment by 200sc** [comment]
As we think this is a shiny bug, this depends on #40

**Comment by 200sc** [comment]
This is potentially fixed on release/2.0.0, I'll close this until we see it reoccur. (It'd be better to have a test that confirmed this, but we can't currently test running oak multiple times to do that, both due to oak's limitations and shiny's limitations)
